tests/playwright/src/model/pages/resource-card-page.ts

```diff
@@ -0,0 +1,49 @@
+/**********************************************************************
+ * Copyright (C) 2024 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import { expect as playExpect, type Locator, type Page } from '@playwright/test';
+
+import { BasePage } from './base-page';
+
+export abstract class ResourceCardPage extends BasePage {
+  readonly parent: Locator;
+  readonly card: Locator;
+  readonly providerSetup: Locator;
+  readonly providerConnections: Locator;
+  readonly markdownContent: Locator;
+
+  constructor(page: Page, resourceName: string) {
+    super(page);
+    this.parent = this.page.getByRole('region', { name: 'content' });
+    this.card = this.parent.getByRole('region', { name: resourceName, exact: true });
+    this.providerSetup = this.card.getByRole('region', { name: 'Provider Setup', exact: true });
+    this.providerConnections = this.card.getByRole('region', { name: 'Provider Connections', exact: true });
+    this.markdownContent = this.providerConnections.getByLabel('markdown-content');
+  }
+
+  public async getSetupButton(buttonLabel: string): Promise<Locator> {
```

Comment:
Hi Anton, there are some issues with this kind of implementation:

1. A method that returns a Locator object does not even need to be a Promise type in reality, since Locator is a an artificial construct of the Playwright framework and it's created synchronously, however you can certainly return a Promise<Locator> for pattern reasons.
2. Getting a locator can be done at any time, even if the actual web element does not exist in the present moment, as such having a method that returns a Locator to asset the object's existence goes against the intended pattern. Asserting existence of an web element should only really be done before that element is actually used.
3. Catching an exception thrown from an assert kind of defeats the purpose, why even use the assert if you are just going to catch it in a try-catch block?